### ISSUE TITLE
event not triggering resolve action from handler while expire_on_resolve = true

### ISSUE THREAD

**Issue by cabecada** [sensu/sensu#1512]
sensu alerts are triggering handler events when action == create.
if the alert is silenced with expire_on_resolve, and alert is resolved, the event gets filtered even with 
expire_on_resolve.

```
#mon_check.json
#handlers are https://github.com/Yelp/sensu_handlers

{
  "checks": {
    "mon_check": {
      "command": "ls /tmp/mon_check",
      "dependencies": [

      ],
      "handlers": [
        "default"
      ],
      "interval": 10,
      "standalone": true,
      "timeout": 10,
      "subdue": {
        "days": {
          "all": [
            {
              "begin": "00:00",
              "end": "00:00"
            }
          ]
        }
      },
      "runbook": "",
      "team": "fake_ops",
      "alert_after": 10,
      "realert_every": "-1",
      "page": true,
      "ticket": false,
    }
  }
}
```

trigger an alert by 
```
touch /tmp/mon_check
sleep 3
rm /tmp/mon_check
```

it rightly triggers the "action=create" event.

now, silence the alert with expire on resolve(from uchiwa)
and verify alert has expire_on_resolve in its silenced data.

```
localhost:6379> mget "silence:client:foobar.com:mon_check"
1) "{\"id\":\"client:foobar.com:mon_check\",\"subscription\":\"client:foobar.com\",\"check\":\"mon_check\",\"reason\":\"pagerduty webhook\",\"creator\":null,\"expire_on_resolve\":true}"
 ```

the alert is silenced and the handler action is now filtered, all good till here.

now
```
touch /tmp/mon_check
```

the check is resolved, the redis silenced entry is removed, but the alert does not fire resolve event although alerts gets resolved.
this is painful, coz the new alerts get filtered as earlier alert is still ack'ed in pagerduty(handler).

* Sensu version used: 0.26.4
* Operating System and version (e.g. Ubuntu 14.04):
* Transport and runtime versions (e.g. RabbitMQ 3.5.2 on Erlang R16B03:

**Comment by cabecada** [comment]
and this is only with expire_on_resolve.
handlers work fine (create and resolve trigger) without expire_on_resolve.

**Comment by portertech** [comment]
Hi @cabecada, thank you for reporting this. We think that the behaviour that you desire is appropriate, I have added this to the 0.28 milestone.

**Comment by cabecada** [comment]
Thanks @portertech am i right in thinking that event's silenced flag is not cleared [here](https://github.com/sensu/sensu/blob/bb91fea7797d2402349a3e86b7cd3f43b78621c8/lib/sensu/server/process.rb#L461)
 it should also set 
```
event[:silenced] = false
```
after it checks expire_on_resolve == true and action == resolve
so that 
when [process_event](https://github.com/sensu/sensu/blob/bb91fea7797d2402349a3e86b7cd3f43b78621c8/lib/sensu/server/process.rb#L253) is called,
[it correctly identifies if the handler is no more silenced in the event](https://github.com/sensu/sensu/blob/bb91fea7797d2402349a3e86b7cd3f43b78621c8/lib/sensu/server/filter.rb#L297) and [and it is filtered to the handler.](https://github.com/sensu/sensu/blob/bb91fea7797d2402349a3e86b7cd3f43b78621c8/lib/sensu/server/filter.rb#L297)

**Comment by cabecada** [comment]
also is it possible to patch this in 0.26.x since this being a minor change.

**Comment by portertech** [comment]
@cabecada https://github.com/sensu/sensu/pull/1519 fixes this, I've reassigned it to the 0.27 milestone, as 0.27 is expected to drop any day now.
